feat: se implementa tests con markers xfail y skip

Ox-Chema-xO opened the pull request:
## Descripcion
Se implementa tests con marcas xfail y skip en casos especificos que dependan de variables de entorno faltantes, errores para determinadas plataformas, funcionalidades aun no implementadas.

## Cambios Realizados
- Se agrega tests con marcas xfail y skip en test_list_utils.py 
- Se agrega tests con marcas xfail y skip en test_string_utils_con_mock.py

Esta PR completa la Issue #23

Ox-Chema-xO committed:
feat(test): agrega test con @pytest.mark.skip

incluye tests de string_utils y list_utils marcados como skip porque no se requiere su ejecucion por ahora

Ox-Chema-xO committed:
feat(test): agrega test con @pytest.mark.xfail

incluye tests de string_utils y list_utils marcados como xfail porque no se implementa la funcionalidad o falla en determinada plataforma

GuidoCh23 reviewed (approved):
Implemenentacion correcta de los markers skip y xfail, no hay ningun failed al ejecutar los test, buen trabajo

GuidoCh23 reviewed (changes requested):
Observacion en la implementacion de xfail en test_string_utils_con_mock.py, anteriormente en windows era estricto para 260 caracteres como esta en el test pero sin embargo ahora a partir de windows 10 en adelante se puede habilitar rutas mas largas de 260, correguir test

Ox-Chema-xO committed:
feat(test): se agrega test con @pytest.mark.xfail

abarca string_utils para una funcionalidad no implementada aun, ademas se elimina test no necesario de rutas para windows
